### PR TITLE
inductor: support linear fusion when multi linear using same input

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106300

For ```llama``` model, there has a pattern that multi linear using the same input and input dim > 2:
```input->view->(linear->view->silu, linear->view)```, this PR update the pattern to make the linar+silu can be fused(we first need remove view ops, and then apply fusion patterns).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 1]
```diff
@@ -328,6 +328,30 @@ def forward(self, x, y):
...
+
+            def forward(self, x):
+                return F.silu(self.w1(x)) * self.w2(x)
+
+        mod = M().to(torch.bfloat16).eval()
+        if torch.ops.mkldnn._is_mkldnn_bf16_supported():
+            v = torch.randn(2, 4, 16).to(torch.bfloat16)
+            # 1. view(match_count=4, match_nodes=4).
+            # 2. mm to packed linear(match_count=2, match_nodes=2).
+            # 3. view+ linear+view to linear(match_count=2, match_nodes=6).
```
Comment (jgong5) [inline test/inductor/test_mkldnn_pattern_matcher.py:349]:
```suggestion
            # 3. view+linear+view to linear(match_count=2, match_nodes=6).
```

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 2]
```diff
@@ -328,6 +328,30 @@ def forward(self, x, y):
...
+        class M(torch.nn.Module):
+            def __init__(
+                self,
+            ):
+                super().__init__()
+                self.w1 = torch.nn.Linear(16, 16, bias=False)
+                self.w2 = torch.nn.Linear(16, 16, bias=False)
+
+            def forward(self, x):
+                return F.silu(self.w1(x)) * self.w2(x)
```
Comment (jgong5) [inline test/inductor/test_mkldnn_pattern_matcher.py:342]: if w2 also has a post-op, would the fusion happen too?

Comment (XiaobingSuper) [reply test/inductor/test_mkldnn_pattern_matcher.py:342]: yes

Comment (jgong5) [reply test/inductor/test_mkldnn_pattern_matcher.py:342]: Do we have UT to cover that too? Please add it if not, otherwise, I'm fine with the PR.

Comment (XiaobingSuper) [reply test/inductor/test_mkldnn_pattern_matcher.py]: Now, update this test case to cover that.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]
